Add go.mod to document dependencies.

Simply following the guide revealed the following error on a Windows system.

```
$ go run .
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\app\settings_desktop.go:10:2: missing go.sum entry for module providing package github.com/fsnotify/fsnotify (imported by fyne.io/fyne/app); to add:
        go get fyne.io/fyne/app@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\painter\gl\gl_core.go:11:2: missing go.sum entry for module providing package github.com/go-gl/gl/v3.2-core/gl (imported by fyne.io/fyne/internal/painter/gl); to add:
        go get fyne.io/fyne/internal/painter/gl@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\driver\glfw\clipboard.go:6:2: missing go.sum entry for module providing package github.com/go-gl/glfw/v3.3/glfw (imported by fyne.io/fyne/internal/driver/glfw); to add:
        go get fyne.io/fyne/internal/driver/glfw@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\painter\gl\gl_common.go:8:2: missing go.sum entry for module providing package github.com/goki/freetype (imported by fyne.io/fyne/internal/painter/gl); to add:
        go get fyne.io/fyne/internal/painter/gl@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\painter\font.go:9:2: missing go.sum entry for module providing package github.com/goki/freetype/truetype (imported by fyne.io/fyne/internal/painter); to add:
        go get fyne.io/fyne/internal/painter@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\painter\image.go:18:2: missing go.sum entry for module providing package github.com/srwiley/oksvg (imported by fyne.io/fyne/internal/painter); to add:
        go get fyne.io/fyne/internal/painter@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\painter\draw.go:9:2: missing go.sum entry for module providing package github.com/srwiley/rasterx (imported by fyne.io/fyne/internal/painter); to add:
        go get fyne.io/fyne/internal/painter@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\painter\image.go:20:2: missing go.sum entry for module providing package golang.org/x/image/draw (imported by fyne.io/fyne/internal/painter); to add:
        go get fyne.io/fyne/internal/painter@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\painter\font.go:10:2: missing go.sum entry for module providing package golang.org/x/image/font (imported by fyne.io/fyne/internal/painter); to add:
        go get fyne.io/fyne/internal/painter@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\internal\painter\draw.go:10:2: missing go.sum entry for module providing package golang.org/x/image/math/fixed (imported by fyne.io/fyne/internal/painter); to add:
        go get fyne.io/fyne/internal/painter@v1.4.3
..\..\..\..\go\pkg\mod\fyne.io\fyne@v1.4.3\app\app_windows.go:17:2: missing go.sum entry for module providing package golang.org/x/sys/windows/registry (imported by fyne.io/fyne/app); to add:
        go get fyne.io/fyne/app@v1.4.3
```

🤷  it seems to work; although a longer term fix might be to upgrade to Fyne v2

--- a/go.mod
+++ b/go.mod
@@ -1,0 +1,5 @@
+module github.com/andydotxyz/sudokgo
+
+go 1.16
+
+require fyne.io/fyne v1.4.3 // indirect

